filters/filter.go

```diff
@@ -0,0 +1,62 @@
+package filters
+
+import (
+	"strings"
+)
+
+type FilterStrategy interface {
```

Comment:
same here... remember that in Go the client of the api is the one interested in the definition of the interface, not the provider of the api. YAGNI.